enhancements/single-node-developer-cluster-profile.md

```diff
@@ -0,0 +1,514 @@
+---
+title: single-node-developer-cluster-profile
+authors:
+  - "@rkukura"
+reviewers:
+  - "@deads2k"
+  - "@LalatenduMohanty"
+  - "@wking"
+approvers:
+  - "@derekwaynecarr"
+  - "@smarterclayton"
+creation-date: 2020-09-18
+last-updated: 2020-10-05
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Single Node Developer Cluster Profile
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Add a new single-node-developer [cluster
+profile](https://github.com/openshift/enhancements/blob/master/enhancements/update/cluster-profiles.md)
+that defines the set of OpenShift components, and the configuration
+thereof, applicable to single-node OpenShift Container Platform
+clusters intended for application development as well as for learning
+and exploring OpenShift, but never for production deployment of
+applications. This cluster profile will be utilized in producing the
+Code Ready Containers (CRC) product that runs in a single VM on a
+developer’s workstation or laptop.
+
+In a cluster deployed using this profile, a single node serves as both
+the cluster’s control plane and as a worker node. The cluster profile
+will support the OpenShift components needed for application
+development, including the operator hub and console. In order to
+minimize system resource requirements, the cluster profile will
+exclude production-oriented features such as high availability of the
+control plane components, machine upgradability, and advanced
+telemetry and logging.
+
+This cluster profile will result in a new topology of OpenShift
+components to be officially supported for its intended usage in the
+CRC product. Once the profile is introduced, all teams developing
+OpenShift components will need to be aware of how their components are
+configured in this profile. For many components, this will be simply a
```

Comment:
One model today for operators to observe signal is to consume the install-config on the bootstrap node. Is observation of CVO annotations on bootstrap an existing pattern?

Reply:
I may be missing your point, but as I understand it, https://github.com/openshift/enhancements/pull/200 has the CVO use the include.release.openshift.io annotations to determine which manifests to process. Once a manifest is selected for processing by the CVO, I'm not sure why the CVO or the operator itself would need to further observe the annotations. What am I missing?

Reply:
I will wait for full review until I see an example of phase 2 manifest. But it feels like we are moving the conditions that should be observed/tested in code to the CVO manifests. Phase 1 makes sense exclude what you don't need but my concern is around phase 2 vs using an explicit `installType` [1].

```
type InstallType string
[..]
type CodeReadyContainer InstallType = "CodeReadyContainer"
// Developer profile means ...
type Developer InstallType = "Developer"
```

[1] https://github.com/openshift/installer/pull/4209